org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/persistence/DocumentProvider.xtend

```diff
@@ -78,6 +84,26 @@ class DocumentProvider {
 		}
 	}
 
+	def boolean isBinary(String resourcePath) {
+		val file = getWorkspaceFile(resourcePath)
+		if (file.exists) {
+			return !java.nio.file.Files.probeContentType(file.toPath).toLowerCase.startsWith("text")
```

Comment:
why not use 'getType' reduce code duplication
how about a new method that checks the mime type to be some text based resource

Reply:
we might also discuss whitelist/blacklist behaviour here. e.g. application/json is not binary

Reply:
thus naming it regardAsBinary might be a better name

Reply:
On using getType: I did consider this, but that method accepts the resourcePath (as it should, its part of the class's public API), and needs to call getWorkspaceFile – again. Now, we could extract another private method that accepts a File or a Path, but then we'd have at least two methods with only a single statement in it, only to avoid the duplication of a single statement. I think that's overkill.

Regarding the other points: agreed!